### ISSUE TITLE
Disco Pane's AddonCompatibilityError component does not need Redux

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#6331]
I was about to fix #6323 and took a look at the component used in Disco Pane (since one suggestion [here](https://github.com/mozilla/addons-frontend/issues/6240#issuecomment-423125561) was about reusing the message from Disco Pane).

I found a component that requires redux but does not use it. We can simplify this component a lot! (and add Flow types).

For QA: disco pane incompatibility errors should be displayed.

**Comment by ioanarusiczki** [comment]
@willdurand 
I verified that "This add-on does not support your browser" message will be shown for the incompatible add-ons (I tried to cover all situations): those who require restart,  not compatible with your version of Firefox,  require a newer version of Firefox,  not available on your platform. 
Tested on AMO dev with FF62(Win10), FF64 (Win10). 

The red warning 

![incompatible with warning](https://user-images.githubusercontent.com/33448286/46006763-f1ee9a00-c0c0-11e8-821f-ede47b19fdfd.png)

I've noticed that the lang packs from the list [ here](https://addons-dev.allizom.org/en-US/firefox/language-tools/) don't display the warning message. I tried with Aragonese and Acholi from the same list of langpacks and I got the same result.

![armenian](https://user-images.githubusercontent.com/33448286/46006425-2e6dc600-c0c0-11e8-82d3-594b33862ee0.png)

**Comment by ioanarusiczki** [comment]
@willdurand Yes, these are not compatible with FF62, the max version as I see for the Armenian is FF53, the reason why with FF62 install button is disabled and the warning message "This add-on is not compatible with your version of Firefox." is displayed on their detail page. 

But this lang pack https://addons-dev.allizom.org/en-US/firefox/addon/acholi-64-language-pack/ has the minimum version 64.0 and maximum version  64.* 
In this case, the disco pane will show the red warning message with FF62 
![max version](https://user-images.githubusercontent.com/33448286/46011347-37fe2a80-c0ce-11e8-9000-5d992c0773cd.png)

**Comment by willdurand** [comment]
> @willdurand Yes, these are not compatible with FF62, the max version as I see for the Armenian is FF53, the reason why with FF62 install button is disabled and the warning message "This add-on is not compatible with your version of Firefox." is displayed on their detail page.

Yep, I see it too now. Thanks! Can you file a new issue for that?

We need an API change because the `addon.current_version.is_strict_compatibility_enabled` field is not exposed in the discovery API response, and this is the field used to ignore/not ignore the max versions.

**Comment by ioanarusiczki** [comment]
I filed an issue here  https://github.com/mozilla/addons-frontend/issues/6401
And closing this one - verified fixed on AMO dev with FF62(Win10), FF64 (Win10).
